tidb-in-kubernetes/k8s-deployment.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Kubernetes 上的 TiDB 集群环境需求
+category: how-to
+---
+
+# Kubernetes 上的 TiDB 集群环境需求
+
+本文介绍在 Kubernetes 上部署 TiDB 集群的软硬件环境需求。
+
+## 软件版本要求
+
+| 软件名称 | 版本 |
+| :--- | :--- |
+| Docker | Docker CE 18.09.6 |
+| Kubernetes |  v1.12.2 |
+| CentOS |  7.6 and kernel 3.10.0-957 or later |
+
+## 内核参数设置
+
+| 配置项 | 设置值 |
+| :--- | :--- |
+| net.core.somaxconn | 32768 |
+| vm.swappiness | 0 |
+| net.ipv4.tcp_syncookies | 1 |
+| net.ipv4.ip_forward | 1 |
+| fs.file-max | 1000000 |
+| fs.inotify.max_user_watches | 1048576 |
+| fs.inotify.max_user_instances | 1024 |
+| net.ipv4.neigh.default.gc_thresh1 | 80000 |
+| net.ipv4.neigh.default.gc_thresh2 | 90000 |
+| net.ipv4.neigh.default.gc_thresh3 | 100000 |
+| net.bridge.bridge-nf-call-iptables | 1 |
+| net.bridge.bridge-nf-call-arptables | 1 |
+| net.bridge.bridge-nf-call-ip6tables | 1 |
+
+在设置 `net.bridge.bridge-nf-call-*` 这几项参数时如果报选项不存在错误，通过如下命令检查是否已经加载改模块：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+lsmod|grep br_netfilter
+```
+
+如果没有加载，则执行如下命令加载：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+modprobe br_netfilter
+```
+
+同时还需要关闭每个部署 Kubernetes 节点的 swap，执行如下命令：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+swapofff -a
+```
+
+执行如下命令检查 swap 是否已经关闭：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+## 执行命令后输出显示 swap 一列全是 0 则表明 swap 已经关闭
+free -m
+```
+
+此外，为了永久性地关闭 swap，还需要将 `/etc/fstab` 中 swap 相关的条目全部删除。
+
+## 硬件和部署要求
+
+与部署 TiDB binary 集群的服务器要求一致，选用 Intel x86-64 架构的 64 位通用硬件服务器，使用万兆网卡。关于 TiDB 集群在物理机上的具体部署需求，参考[这里](/dev/how-to/deploy/hardware-recommendations.md)。
+
+对于服务器 disk、memory、CPU 的选择要根据对集群的容量规划以及部署拓扑来定。线上 Kubernetes 集群部署为了保证高可用一般需要部署三个 master 节点，三个 etcd 节点，若干个 work 节点。同时，为了充分利用机器资源，master 节点一般也充当 work 节点（也就是 master 节点上也可以调度负载），通过 kubelet 设置[预留资源](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/) 来保证机器上的系统进程以及 Kubernetes 的核心进程在工作负载很高的情况下仍然有足够的资源来运行，从而保证整个系统的稳定。
```

Comment:
@onlymellb  I want to confirm: is it `work` or `worker`?

Reply:
I will change to use worker instead